src/App.tsx

```diff
@@ -1,26 +1,133 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState, useRef } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID } from './api/todos';
+import * as todosApi from './api/todos';
+import { Todo } from './types/Todo';
+import { Error } from './components/Error';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { TempTodo } from './components/TempTodo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([] as Todo[]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([] as Todo[]);
```

Comment:
Don't need to create addtional state for filtered todos, you can save filtered todos in the variable
```suggestion
  const [todos, setTodos] = useState<Todo[]>([]);
```